### DISCUSSION TITLE
Performance for ResponsiveSwarmPlot with Many Points and High `simulationIterations`

### DISCUSSION THREAD

**Discussion by lumoe** [plouc/nivo#2595] [Q&A]
I am using the ResponsiveSwarmPlot with about 3000 data points. To prevent overlapping bubbles, I need around 500 simulationIterations. This makes the browser unresponsive due to the high computational load.

Does anyone have suggestions for improving performance in this situation? Reducing the number of points is not an option because the _"large blob"_ is important for storytelling.

Any advice on optimizing performance while keeping the visual representation intact would be greatly appreciated.

Thank you!

**Comment by plouc** [comment]
Hi @lumoe,

Would you be able to provide a codesandbox demonstrating the issue?
I have no doubts that you're facing this issue, but having a simple example to experiment with would help.
That being said, if the issue is mostly in the D3 algorithm, I'm afraid we won't be able to fix it.
